cmd/edit/ingress/cmd.go

```diff
@@ -423,41 +392,70 @@ func run(cmd *cobra.Command, argv []string) {
 				namespaceOwnershipPolicy = &namespaceOwnershipPolicyArg
 			}
 		}
+
+		// old flags for unified component routes
+		// can only set via arguments to allow changing back
+		// interactive mode removed as it's not supported anymore
 		if cmd.Flags().Changed(clusterRoutesHostnameFlag) {
 			if ocm.IsHyperShiftCluster(cluster) {
 				r.Reporter.Errorf("Updating Cluster Routes Hostname is not supported for Hosted Control Plane clusters")
 				os.Exit(1)
 			}
 			clusterRoutesHostname = &args.clusterRoutesHostname
-		} else if interactive.Enabled() && !ocm.IsHyperShiftCluster(cluster) {
-			clusterRoutesHostnameArg, err := interactive.GetString(interactive.Input{
-				Question: "Cluster Routes Hostname",
-				Help:     cmd.Flags().Lookup(clusterRoutesHostnameFlag).Usage,
-				Default:  args.clusterRoutesHostname,
-			})
-			if err != nil {
-				r.Reporter.Errorf("Expected a valid Cluster Routes Hostname: %s", err)
-				os.Exit(1)
-			}
-			clusterRoutesHostname = &clusterRoutesHostnameArg
 		}
 		if cmd.Flags().Changed(clusterRoutesTlsSecretRefFlag) {
 			if ocm.IsHyperShiftCluster(cluster) {
-				r.Reporter.Errorf("Updating Cluster Routes Hostname is not supported for Hosted Control Plane clusters")
+				r.Reporter.Errorf(
+					"Updating Cluster Routes TLS Secret Reference is not supported for Hosted Control Plane clusters",
+				)
 				os.Exit(1)
 			}
 			clusterRoutesTlsSecretRef = &args.clusterRoutesTlsSecretRef
-		} else if interactive.Enabled() && !ocm.IsHyperShiftCluster(cluster) {
-			clusterRoutesTlsSecretRefArg, err := interactive.GetString(interactive.Input{
-				Question: "Cluster Routes TLS Secret Reference",
-				Help:     cmd.Flags().Lookup(clusterRoutesTlsSecretRefFlag).Usage,
-				Default:  args.clusterRoutesTlsSecretRef,
-			})
+		}
+
+		if cmd.Flags().Changed(componentRoutesFlag) {
+			if ocm.IsHyperShiftCluster(cluster) {
+				r.Reporter.Errorf(
+					"Updating Cluster Component Routes is not supported for Hosted Control Plane clusters",
+				)
+				os.Exit(1)
+			}
+			componentRoutes, err = parseComponentRoutes(args.componentRoutes)
 			if err != nil {
-				r.Reporter.Errorf("Expected a valid Cluster Routes TLS Secret Reference: %s", err)
+				r.Reporter.Errorf("An error occurred whilst parsing the supplied component routes: %s", err)
 				os.Exit(1)
 			}
-			clusterRoutesTlsSecretRef = &clusterRoutesTlsSecretRefArg
+		} else if interactive.Enabled() && !ocm.IsHyperShiftCluster(cluster) {
+			componentRoutes = map[string]*cmv1.ComponentRouteBuilder{}
+			if confirm.Prompt(false, "Would you like to edit the component routes?") {
```

Comment:
Is there a way to simplify the `if else` conditions? is it worth extracting some of the logic to new functions?

Reply:
hmm, can only really setup a variable for it. Can't think of a different way at the moment :thinking: